src/App.ts

```diff
@@ -6,84 +6,158 @@
 import * as bodyParser from "body-parser";
 import * as express from "express";
 import * as core from "express-serve-static-core";
+import * as fs from "fs";
 import * as http from "http";
-import { clients as pagoPaSoapClient } from "italia-pagopa-api";
-import { PPTPortTypes } from "italia-pagopa-api/dist/wsdl-lib/PagamentiTelematiciPspNodoservice/PPTPort";
+import { clients as pagoPASoapClient } from "italia-pagopa-api";
+import { FespCdService_WSDL_PATH } from "italia-pagopa-api/dist/lib/wsdl-paths";
+import {
+  IcdInfoPagamentoInput,
+  IcdInfoPagamentoOutput
+} from "italia-pagopa-api/dist/wsdl-lib/FespCdService/FespCdPortType";
 import { toExpressHandler } from "italia-ts-commons/lib/express";
-import { Configuration } from "./Configuration";
+import * as morgan from "morgan";
+import * as redis from "redis";
+import * as soap from "soap";
+import { specs as publicApiV1Specs } from "./api/public_api_pagopa";
+import { CONFIG, Configuration } from "./Configuration";
+import { GetOpenapi } from "./controllers/openapi";
 import * as PaymentController from "./controllers/restful/PaymentController";
 import { logger } from "./utils/Logger";
 
-// Define and start a WS SOAP\Restful Server
+/**
+ * Define and start an express Server
+ * to expose RESTful and SOAP endpoints for BackendApp and Proxy requests.
+ * @param {Configuration} config - The server configuration to use
+ * @return {Promise<http.Server>} The express server defined and started
+ */
 export async function startApp(config: Configuration): Promise<http.Server> {
-  logger.info("Starting Proxy PagoPa Server...");
+  logger.info("Starting Proxy PagoPA Server...");
 
-  // Define SOAP Clients necessary to send messages to PagoPa SOAP WS
-  const verificaRPTPagoPaClient = new pagoPaSoapClient.PagamentiTelematiciPspNodoAsyncClient(
-    await pagoPaSoapClient.createPagamentiTelematiciPspNodoClient({
+  // Define SOAP Clients for PagoPA SOAP WS
+  // It is necessary to forward BackendApp requests to PagoPA
+  const pagoPAClient = new pagoPASoapClient.PagamentiTelematiciPspNodoAsyncClient(
+    await pagoPASoapClient.createPagamentiTelematiciPspNodoClient({
       endpoint: `${config.PAGOPA.HOST}:${config.PAGOPA.PORT}${
-        config.PAGOPA.SERVICES.PAYMENTS_CHECK
-      }`,
-      envelopeKey: PPTPortTypes.envelopeKey
+        config.PAGOPA.WS_SERVICES.PAGAMENTI
+      }?wsdl`
     })
   );
-  const attivaRPTPagoPaClient = new pagoPaSoapClient.PagamentiTelematiciPspNodoAsyncClient(
-    await pagoPaSoapClient.createPagamentiTelematiciPspNodoClient({
-      endpoint: `${config.PAGOPA.HOST}:${config.PAGOPA.PORT}${
-        config.PAGOPA.SERVICES.PAYMENTS_ACTIVATION
-      }`,
-      envelopeKey: PPTPortTypes.envelopeKey
-    })
+
+  // Define a redis client necessary to handle persistent data
+  // for async payment activation process
+  const redisClient = redis.createClient(
```

Comment:
you need to support a redis cluster too (we use it in production), see [this code](https://github.com/teamdigitale/italia-backend/blob/master/src/container.ts#L211)